ethcore/evm/src/interpreter/mod.rs

```diff
@@ -222,23 +222,69 @@ impl<Cost: 'static + CostType> vm::Exec for Interpreter<Cost> {
+            let mut timer: howlong::timer::SteadyTimer;
+            let mut timer_duration: u128;
+            let mut opcode_duration: u128;
+            let mut timer_durations: Vec<u128>;
+            let mut opcode_durations: Vec<u128>;
+
+            // enough to always fit all measurements
+            timer_durations = Vec::with_capacity(200000);
+            opcode_durations = Vec::with_capacity(200000);
+            
+            timer = howlong::timer::SteadyTimer::new();
+            // NOTE: don't use `timer.start();`, counterintuitively, it's much worse
             loop {
-                let timer: howlong::timer::SteadyTimer;
-                let time;
-                timer = howlong::timer::SteadyTimer::new();
                 let result = self.step(ext);
-                time = timer.elapsed().as_nanos();
-                println!("{:?},{:?},{:?}", sample, instruction_number, time);
-                instruction_number += 1;
                 match result {
-                    InterpreterResult::Continue => {}
-                    InterpreterResult::Done(value) => return Ok(value),
+                    InterpreterResult::Continue => {
+                        opcode_duration = timer.elapsed().as_nanos();
+                        timer = howlong::timer::SteadyTimer::new();
+                        timer_duration = timer.elapsed().as_nanos();
+
+                        timer_durations.push(timer_duration);
+                        opcode_durations.push(opcode_duration);
+                        
+                        timer = howlong::timer::SteadyTimer::new();
+                    }
+                    InterpreterResult::Done(value) => {
+                        opcode_duration = timer.elapsed().as_nanos();
+                        timer = howlong::timer::SteadyTimer::new();
+                        timer_duration = timer.elapsed().as_nanos();
+
+                        timer_durations.push(timer_duration);
+                        opcode_durations.push(opcode_duration);
+
+                        for n in 0..timer_durations.len() {
+                            println!("{:?},{:?},{:?},{:?}", sample, n, opcode_durations[n], timer_durations[n]);
+                        }
+                        return Ok(value)
+                    }
                     InterpreterResult::Trap(trap) => match trap {
                         TrapKind::Call(params) => {
+                            opcode_duration = timer.elapsed().as_nanos();
+                            timer = howlong::timer::SteadyTimer::new();
+                            timer_duration = timer.elapsed().as_nanos();
+
+                            timer_durations.push(timer_duration);
+                            opcode_durations.push(opcode_duration);
+
+                            for n in 0..timer_durations.len() {
+                                println!("{:?},{:?},{:?},{:?}", sample, n, opcode_durations[n], timer_durations[n]);
+                            }
```

Comment:
Can we lift it to a func? Maybe in this code we ensure that there is a little additional calls as possible, but the code is unfortunately copy-pasted in 4 places, which seems reasonable to extract it to become a func.

Reply:
@pik694 :blush: I'd need to learn more rust... it takes several minutes to build this thing... do we have to :) ?

I'll give it a try, but time cap it to say 10 mins. If you know how to, pls lmk and just commit the change

Reply:
You can leave it to me then 😉

Reply:
oh @pik694 but please don't move the capturing of time to the function, only the pieces that are "outside" of the measured execution!

The 
```rust
                             opcode_duration = timer.elapsed().as_nanos();
                            timer = howlong::timer::SteadyTimer::new();
                            timer_duration = timer.elapsed().as_nanos(); 
```

should stay where it is :(

Reply:
we could just use an inlined function like this:
```line(always)]
fn measure(
    timer: &mut howlong::timer::SteadyTimer,
    opcode_duration: &mut u128,
    timer_duration: &mut u128,
    timer_durations: &mut Vec<u128>,
    opcode_durations: &mut Vec<u128>,
) {
    *opcode_duration = timer.elapsed().as_nanos();
    *timer = howlong::timer::SteadyTimer::new();
    *timer_duration = timer.elapsed().as_nanos();

    timer_durations.push(*timer_duration);
    opcode_durations.push(*opcode_duration);
}```
  But I'm not quite sure if we should do it and hope that the compiler would care to listen our hints - though the doc says it should: https://nnethercote.github.io/perf-book/inlining.html